### PR TITLE
Allows overriding of default formatters and plugins

### PR DESCRIPTION
**Pull request by DavidTobin**
This PR allows for overriding of default plugins and formatters. It extends the current behavior of allowing command patches to be overridden.

### CODE REVIEW

File: src/scribe.js [hunk 1]
```diff
@@ -109,27 +115,40 @@ define([
...
     if (this.allowsBlockElements()) {
-      // Commands assume block elements are allowed, so all we have to do is
-      // set the content.
-      // TODO: replace this by initial formatter application?
-      this.use(setRootPElement());
-      // Warning: enforcePElements must come before ensureSelectableContainers
-      this.use(enforcePElements());
-      this.use(ensureSelectableContainers());
+      /**
+       * Warning: enforcePElements must come before ensureSelectableContainers
```
Comment (rrees) [inline src/scribe.js]: I think we should keep this rule enforced in code

**Comment by rrees** [comment]
This looks absolutely smashing, thanks!

Could I be a bit cheeky and ask whether I could merge #356 and then get you to add the defaults to that model? I was worried that main Scribe was getting a bit bloated.

**Comment by DavidTobin** [comment]
Of course, work away!

File: src/scribe.js (hunk 1 above)
Comment (DavidTobin) [inline src/scribe.js]: Will update!

**Comment by DavidTobin** [comment]
@rrees Rebased with current master and made a change to force `setRootPElement` to load first. How does this look now?

**Comment by rrees** [comment]
@DavidTobin I'm thinking this would be a great use case for ImmutableJS's OrderedSets!

In terms of this change though, could you extract the sort function into a named entity and add a unit-test for its behaviour?

I worried that the inline sort isn't that clear from a maintenance point of view.

Apart from that, good to go.

**Comment by DavidTobin** [comment]
Makes total sense! I'll update soon.

**Comment by DavidTobin** [comment]
Updated. Let me know if anything else is out of place.

**Comment by rrees** [comment]
Okay so this is failing consistently on Firefox so I'll revert and we can try and figure out what is going on.

**Comment by DavidTobin** [comment]
Looks to be a poor assertion in my override test.

**Comment by rrees** [comment]
Can you raise a new pull request?

**Comment by DavidTobin** [comment]
Will do.

**Pull request merged**